website/blog/2018-06-19-single-spa-parcels-explained.md

```diff
@@ -0,0 +1,70 @@
+---
+title: single-spa parcels, explained
+author: Joel Denning
+authorURL: https://twitter.com/joelbdenning
+authorImageURL: https://avatars2.githubusercontent.com/u/5524384?s=400&u=ff145fcb2ae5305555628a446e9f725d4e145aaa&v=4
+---
+Ever since single-spa@1.0.0, the single-spa team has been dedicated to bringing microservices to the frontend. We have made it possible for AngularJS, React, Angular, Vue, and other frameworks to coexist side by side in the same page.
+
+And with the release of [version 4](https://github.com/CanopyTax/single-spa/releases/tag/v4.0.0), I’m pleased to announce that [single-spa](https://github.com/CanopyTax/single-spa) is expanding that effort so that ***individual components*** written with different frameworks can interoperate. It is new terrain for the single-spa community, which previously had focused on getting large applications to interoperate with each other, instead of the individual components.
+
+## Another way to do framework agnostic components?
+
+For those familiar with [web components](https://developer.mozilla.org/en-US/docs/Web/Web_Components) and [custom elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements), you may be wondering why a javascript library would try to do what browsers are starting natively to do.
+
+And as one of the contributors to the custom elements polyfill, let me be the first one to say that we did not make this decision lightly.
+
+If you’re interested in diving into the details, check out [One Company’s Relationship With Custom Elements](https://medium.com/canopy-tax/one-companys-relationship-with-custom-elements-d360baf3b253), which explains some of the difficulties we’ve been through with web components and custom elements.
+
+TLDR: React and some other frameworks don’t interop with custom elements very well. Additionally dealing with inner html, attributes vs properties, and customized builtins can be a pain.
+
+## Okay but you haven’t told me what a single-spa parcel is
+
+A parcel is single-spa’s way of building a component in one framework and using it in another.
+
+To implement a parcel, just create a javascript object that has 3–4 functions on it. We call this javascript object a *parcel config* and there are three required functions to implement: bootstrap, mount, and unmount. A fourth function, update, is optional.
```

Comment:
Possibly link to the docs here so people can read more about what the bootstrap, mount and unmount functions are doing.

Reply:
Same as above - this is a copy/paste job from medium and i'm not really wanting to have two different versions of the blog post.